### PR TITLE
Fix #2423: Mathematical Naming in classify rules

### LINKED ISSUE #2423: Fix Mathematical Naming in classify rules
**Describe the bug**
Some of the variables naming related to numbers in `src/main/java/org/oppia/android/domain/classify/rules` and it's test packet are mathematical incorrect. 
example - `NEGATIVE_WHOLE_NUMBER_VALUE_TEST_2` where a whole number cannot be negative. 

**Expected behavior**
Need to fix all such variables name

Follow these comments to solve this issue - https://github.com/oppia/oppia-android/pull/2443#issuecomment-758299254
Reach out to @anandwana001 before start working on this. 

### How to work on this issue
Suggesting few points below on how to solve this particular issue:

1. Let's have a doc where you can list all the types of the number we have.
2. Specify the rules based on those numbers. Example - A whole number can't be negative.
3. You can add rules for all the types of number like a fraction, mixed fraction, Ratio of numbers etc...
4. After completing with doc, either we can fix everything in this PR, but I guess it might make this PR big, so we can divide it into multiple sub-issues.

### PR DESCRIPTION
**Pull request by yash10019coder**
Fixes #2423 
changed NEGATIVE_WHOLE_NUMBER variable to NEGATIVE_INTEGER_NUMBER

changed function naming in InteractionObjectTestBuilder.kt

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by Sparsh1212 — CHANGES_REQUESTED** [review]
Thanks, @yash10019coder for the fix!

Can you please link the issue number which this PR fixes, and also mark all the points in the checklist.
I also suggest, checking the formatting of the files. Because currently, it looks like, the indentations are not correct.

**Comment by yash10019coder** [comment]
How to link issue to PR?

**Comment by prayutsu** [comment]
Hi @yash10019coder 
Can you please add the prefix "Fix #issueNumber" to the title of the PR as per our [PR guidelines](https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change).
Also, you should add the PR explanation below the `Explanation` heading.
You should include only lowercase English letters in the branch name and words should be separated by '-'.
You can take a look at other PRs for reference.

**Comment by prayutsu** [comment]
@yash10019coder Also it looks like you have linked a wrong issue. Please correct the issue Number after the "Fixes" keyowrd.

File: domain/src/test/java/org/oppia/android/domain/classify/InteractionObjectTestBuilder.kt [hunk 1]
```diff
@@ -72,7 +72,7 @@ object InteractionObjectTestBuilder {
       }.build()
   }.build()
 
-  fun createWholeNumber(isNegative: Boolean, value: Int): InteractionObject {
+  fun createWholeNumber(isNegative: Boolean = false, value: Int): InteractionObject {
     // Whole number fractions imply '0/1' fractional parts.
```
Comment (prayutsu) [inline domain/src/test/java/org/oppia/android/domain/classify/InteractionObjectTestBuilder.kt:76]: We'll probably want to remove the parameter `isNegative` from this method, as there can't be a negative whole number.
Adding @vinitamurthi for confirmation.

**Review by prayutsu — CHANGES_REQUESTED** [review]
@yash10019coder There are also other files where we want to incorporate these changes. Can you make changes in other files too?

File: domain/src/test/java/org/oppia/android/domain/classify/InteractionObjectTestBuilder.kt (hunk 1 above)
Comment (yash10019coder) [reply domain/src/test/java/org/oppia/android/domain/classify/InteractionObjectTestBuilder.kt:76]: Yes @prayutsu

File: domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsEquivalentToAndInSimplestFormRuleClassifierProviderTest.kt [hunk 2]
```diff
@@ -32,7 +32,7 @@ class FractionInputIsEquivalentToAndInSimplestFormRuleClassifierProviderTest {
       isNegative = false,
       value = 2
     )
-  private val NEGATIVE_WHOLE_NUMBER_VALUE_TEST_2 =
+  private val NEGATIVE_INTEGER_NUMBER_VALUE_TEST_2 =
     InteractionObjectTestBuilder.createWholeNumber(
       isNegative = true,
```
Comment (prayutsu) [inline domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsEquivalentToAndInSimplestFormRuleClassifierProviderTest.kt]: Also, please use a different method here to create Negative integers (you can use `createSignedInt()` method).

Comment (yash10019coder) [reply domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsEquivalentToAndInSimplestFormRuleClassifierProviderTest.kt]: Ok @prayutsu

**Comment by yash10019coder** [comment]
Hi I have made the changes in the PR please check now @Sparsh1212 and @prayutsu

**Comment by anandwana001** [comment]
@yash10019coder Please check the comments here - https://github.com/oppia/oppia-android/pull/2443#issuecomment-758299254
and please provide a doc describing how we are planning to solve this issue. More details are in the comments from the above link.
Feel free to reach out to me if needed.

File: domain/src/test/java/org/oppia/android/domain/classify/AnswerClassificationControllerTest.kt [hunk 3]
```diff
@@ -70,8 +69,8 @@ class AnswerClassificationControllerTest {
     private val TEST_ITEM_SELECTION_SET_1 =
       createSetOfTranslatableHtmlContentIds("content_id_0", "content_id_2", "content_id_3")
 
-    private val TEST_MULTIPLE_CHOICE_OPTION_0 = createNonNegativeInt(value = 0)
-    private val TEST_MULTIPLE_CHOICE_OPTION_1 = createNonNegativeInt(value = 1)
+    private val TEST_MULTIPLE_CHOICE_OPTION_0 = createUnsingnedInteger(value = 0)
+    private val TEST_MULTIPLE_CHOICE_OPTION_1 = createUnsingnedInteger(value = 1)
```
Comment (prayutsu) [inline domain/src/test/java/org/oppia/android/domain/classify/AnswerClassificationControllerTest.kt:74]: @yash10019coder There is a typo in `createUnsignedInteger()`, also I think the original name of the method is more meaningful.

**Review by prayutsu — CHANGES_REQUESTED** [review]
@yash10019coder Left one comment. PTAL.

File: domain/src/test/java/org/oppia/android/domain/classify/AnswerClassificationControllerTest.kt (hunk 3 above)
Comment (yash10019coder) [reply domain/src/test/java/org/oppia/android/domain/classify/AnswerClassificationControllerTest.kt:74]: So I shall correct the typo or change the function name to the original name @prayutsu

Comment (prayutsu) [reply domain/src/test/java/org/oppia/android/domain/classify/AnswerClassificationControllerTest.kt:74]: You can restore the previous name.

Comment (yash10019coder) [reply domain/src/test/java/org/oppia/android/domain/classify/AnswerClassificationControllerTest.kt:74]: ok @prayutsu

**Comment by anandwana001** [comment]
@yash10019coder are you still working on this?

**Comment by yash10019coder** [comment]
Yes @anandwana001 I'm still working on this

**Comment by yash10019coder** [comment]
@prayutsu @anandwana001 made changes PTAL

**Comment by anandwana001** [comment]
> @yash10019coder Please check the comments here - [#2443 (comment)](https://github.com/oppia/oppia-android/pull/2443#issuecomment-758299254)
> and please provide a doc describing how we are planning to solve this issue. More details are in the comments from the above link.
> Feel free to reach out to me if needed.

@yash10019coder As per my earlier comment, to fix this issue we need more details. 
Could you follow my earlier comments and assign me back once done. 

@prayutsu Before moving forward with this PR in review, please let the contributor follow the steps given in the issue. 

Also, @yash10019coder This issue is going to be a big one, as it involves more documentation than technical stuff. So, feel free to let us know if you want to continue with this issue or you want to work on another issue.  By next I will go through the issue again and bring up some more smooth steps to complete this issue.

File: domain/src/test/java/org/oppia/android/domain/classify/AnswerClassificationControllerTest.kt (hunk 3 above)
Comment (prayutsu) [reply domain/src/test/java/org/oppia/android/domain/classify/AnswerClassificationControllerTest.kt:74]: @yash10019coder The original name is `createNonNegativeInt` and not `createNonNegativeInteger`

**Review by prayutsu — CHANGES_REQUESTED** [review]
Please restore the correct name of the method and fix the failing checks.

**Comment by yash10019coder** [comment]
> > @yash10019coder Please check the comments here - [#2443 (comment)](https://github.com/oppia/oppia-android/pull/2443#issuecomment-758299254)
> > and please provide a doc describing how we are planning to solve this issue. More details are in the comments from the above link.
> > Feel free to reach out to me if needed.
> 
> @yash10019coder As per my earlier comment, to fix this issue we need more details.
> Could you follow my earlier comments and assign me back once done.
> 
> @prayutsu Before moving forward with this PR in review, please let the contributor follow the steps given in the issue.
> 
> Also, @yash10019coder This issue is going to be a big one, as it involves more documentation than technical stuff. So, feel free to let us know if you want to continue with this issue or you want to work on another issue. By next I will go through the issue again and bring up some more smooth steps to complete this issue.

It says that the doc does not exist
[https://docs.google.com/document/d/1ld3wdeS61-_fery79mcCXczdSYJzSFeyi7360DlIR1c/edit?usp=sharing](url)

**Pull request closed without merging**